### DISCUSSION TITLE
Allow multiple backups at a time

### DISCUSSION THREAD

**Discussion by samuel-w** [borgbase/vorta#830] [Ideas]
**Describe the bug**
I cannot run multiple backups at a time, even though they are in different repositories.

**To Reproduce**
1. Start a backup in one profile
2. Go to another profile with a different repository, and try to start a backup
Result: Backup button is greyed out, even though you can run multiple backups at a time from the command line

**Desktop (please complete the following information):**
- OS: Ubuntu 20.04
- Vorta version: 0.7.1
- Borg version: 1.1.11

Vorta and Borg versions can be found in Main Window > Misc Tab

**Additional context**
Must fix for #103

**Comment by m3nu** [comment]
That's correct. There can only be *one* Borg process at a time. The logging and other things are all set up on this assumption. To change it:

- The Borg lock needs to work at the profile level, not globally. (The same repo can be used in multipe profiles so this case needs handling too)
- How to deal with logs and status updates, when multiple processes are running?
- ...

We can keep this issue open, but it's a bigger project for sure.

**Comment by samuel-w** [comment]
For checking borg lock we can check if `lock.exclusive` exists in the repo url. Not sure about remote.

Status updates we could separate by profile when multiple are running, so that you could only see the current profile's status.
Logs are thread safe already, https://docs.python.org/3/library/logging.html#thread-safety. We could append the repo url for clarity.

**Comment by samuel-w** [comment]
As an alternative, we could implement multiple backups as a queue. It would be easier, just run the next in queue after the current ends.

**Comment by olivierlemoal** [comment]
I started drafting thread locks at repository level https://github.com/olivierlemoal/vorta/tree/feature/enable-parallel-backup
In the current state :
- Threads related to a repository won't start if the repository is already in use. It is allowed to start otherwise.
- Backups can be run simultaneously from UI / CLI args / Scheduling
- To be able to see all notifications on desktop, previous notification is not replaced anymore by any additional notification, but all notifications are displayed to users (on Linux)
- Progress text and logs on UI are updated accordingly to selected profile

But : 
- logging is a bit messy as it is not descriptive enough on which profile it relates
- It might be nice to have an option to allow/prevent multiple backup in // (limited bandwidth...)
- Queuing is still interesting for multiple backups using the same repo, or if we do not wish to have multiple backups running simultaneously 
- Current code lacks UX :
    - "Cancel" in systray will cancel all backups


Any additional input / review / comment is appreciated !

**Reply by bastiencyr** [reply]
Hello @olivierlemoal , 

> Queuing is still interesting  (...) or if we do not wish to have multiple backups running simultaneously

We can lunch multiple backup with mutliple queues. 

I think the queue is more efficient. It will solve  #263 (schedule at next opportunity), this issue,  and will help me on  #1005 to properly implement multiple repos from one profile. Maybe we can implement threads locks at repository level and multiple  queues. But several queue is enough ?
Currently, I plan to understand #908  since it is suppose to solve #263
